main.tf

```diff
@@ -137,15 +137,19 @@ resource "random_id" "cloudnat_suffix" {
 }
 
 module "cloud_nat" {
-  source           = "terraform-google-modules/cloud-nat/google"
-  version          = "~> 2.2.0"
-  project_id       = var.project_id
-  region           = var.region
-  router           = format("%s-router", var.project_id)
-  name             = "${var.project_id}-cloud-nat-${random_id.cloudnat_suffix.hex}"
-  network          = google_compute_network.gitlab.self_link
-  create_router    = true
-  min_ports_per_vm = "2048"
+  source        = "terraform-google-modules/cloud-nat/google"
+  version       = "~> 2.2.0"
+  project_id    = var.project_id
+  region        = var.region
+  router        = format("%s-router", var.project_id)
+  name          = "${var.project_id}-cloud-nat-${random_id.cloudnat_suffix.hex}"
+  network       = google_compute_network.gitlab.self_link
+  create_router = true
+  # We force the endpoint independent mapping to false as described in this issue:
+  # https://github.com/hashicorp/terraform-provider-google/issues/10609
+  enable_endpoint_independent_mapping = false
+  min_ports_per_vm                    = "4096"
+  log_config_enable                   = true
```

Comment:
make those variables configurable and stick with the default values of the cloud-nat module, especially the logging it's very opinionated and the ports too. 
It seems that the default min_ports_per_vm is 64 (https://registry.terraform.io/modules/terraform-google-modules/cloud-nat/google/latest) we should check better those values.

Reply:
Fixed.